### PR TITLE
HHH-13685 Upgrade to Gradle 5

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HHH-13685

Before we merge this, we need to upgrade and release three Gradle plugins:

* ~https://github.com/sebersole/gradle-maven-publish-auth/pull/20~ => Done
* ~https://github.com/hibernate/hibernate-matrix-testing/pull/9~ => Done
* ~https://github.com/sebersole/gradle-xjc-plugin/pull/3~ => Done

I checked:

* publishing to the local Maven repository
* publishing snapshots to nexus
* running the matrix plugin

I'd like to run tests on CI before we merge, but we need to release the Gradle plugins first.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -84,14 +84,13 @@ task ciBuild {
...
 wrapper {
-	gradleVersion = '4.10.3'
 	distributionType = Wrapper.DistributionType.ALL
 }
 
 
 buildScan {
-	licenseAgreementUrl = 'https://gradle.com/terms-of-service'
-	licenseAgree = 'yes'
+	termsOfServiceUrl = "https://gradle.com/terms-of-service"
```
Comment (Sanne) [inline build.gradle]: it's best to use the single quotes in this case, as it's a constant. the double quotes enable formatting / interpolation of variables.

Comment (yrodiere) [reply build.gradle]: Ok; done

**Comment by yrodiere** [comment]
@sebersole Done. I had to upgrade the `maven-publish-auth` plugin, I added a commit here and submitted a few PRs in the appropriate repos:

* sebersole/gradle-maven-publish-auth#20
* hibernate/hibernate-matrix-testing#9
* sebersole/gradle-xjc-plugin#3

I didn't test the bintray release, but I imagine I need credentials that I don't have, and from what I understand it's not for snapshots.

**Comment by yrodiere** [comment]
The Gradle plugins have all been published. I just rebased and force-pushed; the PR now uses the latest version of every plugin.

The build passes locally, and I managed to [publish to bintray](https://bintray.com/yrodiere/hibernate-release-testing/hibernate-orm#files/org/hibernate).

Note that we may have to wait a bit before CI passes, because the matrix plugin has not been sync'ed with Maven Central yet.

**Pull request merged**